.github/workflows/continuous-integration.yaml

```diff
@@ -427,21 +427,81 @@ jobs:
           make actionlint
 
       - name: Source code conformance
-        id: cs
+        id: make-conformance
         if: ${{ steps.install-deps.outcome == 'success' && !cancelled() }}
         # pcc launches an internal target with proper flags
         run: |
           make pcc
 
+      # Checked for changes between main and the current branch in a PR. More specifically, 
+      # this is used in regular CIs to avoid launching Pytest, checking codeblocks, building docs 
+      # or other steps if the associated files were not touched. For most, we also check that the
+      # linux MD5 has not changed, which means that no libraries got updated. This is done in order
+      # to handle PRs which only upgrades dependencies
+      - name: Get all changed files from main in PR
+        id: changed-files-in-pr
+        if: | 
+          env.IS_PR
+          && !fromJSON(env.IS_WEEKLY) 
+          && !fromJSON(env.IS_RELEASE)
+          && steps.install-deps.outcome == 'success'
+          && steps.make-conformance.outcome == 'success' 
+          && !cancelled()
+        uses: tj-actions/changed-files@v37.5.1
+        with:
+          files_yaml: |
+            src:
+              - src/**
+            tests:
+              - tests/**
+              - src/concrete/ml/pytest/**
+              - conftest.py
+              - deps_licenses/licenses_linux_user.txt.md5
+            determinism:
+              - src/**
+              - tests/seeding/test_seeding.py
+              - conftest.py
+              - deps_licenses/licenses_linux_user.txt.md5
+            docs:
+              - docs/**
+            codeblocks:
+              - src/**
+              - '**.md'
+              - '!.*/**'
+              - '!docs/_*/**'
+              - deps_licenses/licenses_linux_user.txt.md5
+            dependencies:
+              - deps_licenses/licenses_linux_user.txt.md5
```

Comment:
Maybe poetry.lock would be better no?

Reply:
mo because the md5 also takes into account the CP version that we install in the makefile), which poetry lock does not ! We thought with @bcm-at-zama that his was simpler and more robust ! But I'll add a comment to make this clear yes